File: python_backend/app/application/use_cases/delete_session.py
"""Soft-delete a session."""
from app.application.ports.sessions_repository import SessionsRepository
from app.domain.exceptions import NotFoundError


class DeleteSessionUseCase:
    def __init__(self, repo: SessionsRepository) -> None:
        self.repo = repo

    def execute(self, *, session_id: str, user_id: str) -> bool:
        deleted = self.repo.soft_delete(session_id, user_id)
        if not deleted:
            raise NotFoundError(f"sesión {session_id} no encontrada")
        return True

File: apps/web/src/application/use-cases/delete-session.ts
/**
 * DeleteSessionUseCase — DELETE /api/sessions/{id}.
 *
 * Soft-deletes the entire session (transcripts + hints + recording + tags).
 * Different from `DeleteRecordingUseCase` which only removes the audio.
 */

import type { SessionsApiPort } from "@/application/ports/sessions-api.port";

export class DeleteSessionUseCase {
  constructor(private readonly api: SessionsApiPort) {}

  execute(sessionId: string): Promise<boolean> {
    return this.api.delete(sessionId);
  }
}
